### ISSUE TITLE
cvtsudoers JSON output isn't well-specified

### ISSUE THREAD

**Issue by zombiepigdragon** [sudo-project/sudo#172]
In [the documentation](https://github.com/sudo-project/sudo/blob/158facf6d5852ebc420adca5ff06135b18ee57b8/docs/cvtsudoers.mdoc.in#L135), it is implied the purpose of `cvtsudoers -f json` is to allow third-party applications to parse sudo rules. However, there does not seem to be any way to get a complete overview of the format without reading the [source](https://github.com/sudo-project/sudo/blob/158facf6d5852ebc420adca5ff06135b18ee57b8/plugins/sudoers/cvtsudoers_json.c) that outputs it or by using trial and error to collect all of the possible keys/values. Am I correct that the JSON output format is meant to be a better-standardized machine-parsable representation of the `sudoers` format? If so, does any additional documentation exist, or where would it go if it was written?

**Comment by millert** [comment]
Yes, the JSON output is intended to be consumed by third-party programs (or converted to XLM and then ingested).  The format is not currently documented, though it is not particularly complicated.  What kind of documentation would you find useful?  A sudoers.json manual page?  A JSON schema file? Something else?

**Comment by zombiepigdragon** [comment]
> What kind of documentation would you find useful? A sudoers.json manual page? A JSON schema file? Something else?

The JSON schema was exactly what I had in mind, actually. I originally expected to find some form of documentation in the `cvtsudoers` manual page, but it hit me later that putting the full schema in there would probably be too long for the pages.

The main things I'm looking for are completeness (I can't be sure if I'm missing a field if I simply convert my desktop's configuration like I first tried) and ideally some short statement of purpose for each field. Unless I'm mistaken, a JSON schema is exactly that, and so its probably the most useful documentation for either the website or a manual page.

An example output being provided somewhere (either in the `cvtsudoers` man page itself or e.g. `/usr/share/doc/cvtsudoers/sudoers.json`) would be useful alongside such a schema, so that the output could be referenced on known-good sample data.

Separately, someone I asked to double check while I was searching for documentation pointed out the csv output format also isn't defined, and having that as well would be icing on top of the JSON format.

**Comment by millert** [comment]
I've added sections to the cvtsudoers man page describing the JSON and CSV output.  I plan to add a JSON schema in the future.
